packages/utils/src/collection/fifo.ts

```diff
@@ -0,0 +1,51 @@
+type Entry<T> = {
+  data: T
+  next: Entry<T> | undefined
+}
+
+/**
+ *
+ * @returns
+ */
+export function FIFO<T>() {
```

Comment:
used in `oneAtATime` to avoid `Array.push()` + `Array.shift()`

Reply:
Throwback to my CS classes where we had to implement LinkedLists manually